### ISSUE TITLE
Native modules used inside plugin fails on Window

### ISSUE THREAD

**Issue by qtphan734** [cypress-io/cypress#4162]
### Current behavior:

Using native modules inside the pluginsfile (`plugins/index.js`) will throw `DLL initialization fail` error on Windows. 

### Desired behavior:

Should be able to use native modules inside plugins file

### Steps to reproduce: (app code and test code)

Inside the pluginsfile, I am requiring the `serialport` module. 

```js
const serialport = require('serialport');
```

Below is the error. Also, Cypress was configured to run in Chrome. 

```
Error: A dynamic link library (DLL) initialization routine failed.
\\?\C:\Code\project\node_modules\@serialport\bindings\build\Release\bindings.node
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:172:20)
    at Object.Module._extensions..node (module.js:598:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:172:20)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
    at Module.require (module.js:513:17)
    at require (internal/module.js:11:18)
    at bindings (C:\Code\project\node_modules\bindings\bindings.js:112:48)
    at Object.<anonymous> (C:\Code\project\node_modules\@serialport\bindings\lib\win32.js:1:191)
    at Object.<anonymous> (C:\Code\project\node_modules\@serialport\bindings\lib\win32.js:111:3)
    at Module._compile (module.js:569:30)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:503:32)
    at tryModuleLoad (module.js:466:12)
    at Function.Module._load (module.js:458:3)
```

### Versions
Cypress Version: `3.1.0`
OS: `Windows 10`
Browser: Chrome `v74.0.3729.131`
Node: `v8.15.1`

**Comment by jennifer-shehane** [comment]
Do you mind my asking why you want to use `serialport` in the plugins?

There seem to be a few issues open with `serialport` and Electron (which we use under the hood at Cypress). https://github.com/serialport/node-serialport/issues/538

We are on electron version 2.0.18 since Cypress 3.3.0 (if you can try updating).

**Comment by qtphan734** [comment]
@jennifer-shehane I am using serialport to control a device before running tests.  However, I believe this apply to any native modules, not just serialport.  Based on my understanding, when you perform npm install, it builds native modules against the Node.js version that was installed on your machine, not that of Electron project (eg. Cypress). 

In my previous projects, I would rebuild all native modules for Electron using the below module:
https://github.com/electron/electron-rebuild

Is there some way this could be implemented in Cypress?

**Comment by JRoppert** [comment]
@qtphan734 Did you find a workaround for that issue? I need to use mmmagic to inspect downloaded images and get the exact same error.
@jennifer-shehane Any idea about how to get around the problem from cypress?

**Comment by qtphan734** [comment]
@JRoppert not yet, still waiting for a solution. for now, cypress is being run only on ubuntu.

**Comment by ddesna** [comment]
I'm using [xml2json](https://github.com/patrikx3/xml2json) inside my custom reporter and I have the same error (I'm also on Windows machine):
```js
Error: A dynamic link library (DLL) initialization routine failed.
\\?\C:\Users\ddesna\git\cypress-custom-reporter\node_modules\node-expat\build\Release\node_expat.node
    at process.module.(anonymous function) [as dlopen] (ELECTRON_ASAR.js:166:20)
    at Object.Module._extensions..node (module.js:671:18)
    at Object.module.(anonymous function) [as .node] (ELECTRON_ASAR.js:166:20)
    at Module.load (module.js:561:32)
    at tryModuleLoad (module.js:504:12)
```

PD:
@qtphan734 I'm impressed about your use case of `serialport` with Cypress 😮

**Comment by flotwig** [comment]
This is most likely because Cypress uses the version of Node bundled with Electron (8.9.3) which is different than the version of node that `node-expat` was installed for.

If you clear `node_modules` and use Node 8.9.3 to do `npm install xml2json`, it should work.

#4436 should resolve this by using your system Node for the custom reporter.

**Comment by jennifer-shehane** [comment]
Released in [`3.5.0`](https://on.cypress.io/changelog#3-5-0).
